source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -2016,4 +2048,90 @@ namespace AdaptiveCards { namespace Uwp
         THROW_IF_FAILED(m_hostConfig->get_SupportsInteractivity(&supportsInteractivity));
         return Boolify(supportsInteractivity);
     }
+
+    void XamlBuilder::WrapInFullWidthTouchTarget(
+        IUIElement* elementToWrap,
+        IAdaptiveActionElement* action,
+        IAdaptiveRenderContext* renderContext,
+        IUIElement** finalElement)
+    {
+        ABI::AdaptiveCards::Uwp::ActionType actionType;
+        THROW_IF_FAILED(action->get_ActionType(&actionType));
+
+        // TODO: In future should support inline ShowCard, but that's complicated for inline elements
+        if (actionType == ABI::AdaptiveCards::Uwp::ActionType::ShowCard)
+        {
+            ComPtr<IAdaptiveActionsConfig> actionsConfig;
+            THROW_IF_FAILED(m_hostConfig->get_Actions(actionsConfig.GetAddressOf()));
+            ComPtr<IAdaptiveShowCardActionConfig> showCardActionConfig;
+            THROW_IF_FAILED(actionsConfig->get_ShowCard(&showCardActionConfig));
+            ABI::AdaptiveCards::Uwp::ActionMode showCardActionMode;
+            THROW_IF_FAILED(showCardActionConfig->get_ActionMode(&showCardActionMode));
+            if (showCardActionMode == ABI::AdaptiveCards::Uwp::ActionMode::Inline)
+            {
+                // Was inline show card, so don't wrap the element and just return
+                ComPtr<IUIElement> localElementToWrap(elementToWrap);
+                localElementToWrap.CopyTo(finalElement);
+                return;
+            }
+        }
+
+        ComPtr<IButton> button = XamlHelpers::CreateXamlClass<IButton>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Button));
+
+        ComPtr<IContentControl> buttonAsContentControl;
+        THROW_IF_FAILED(button.As(&buttonAsContentControl));
+        THROW_IF_FAILED(buttonAsContentControl->put_Content(elementToWrap));
+
+        ComPtr<IAdaptiveSpacingConfig> spacingConfig;
+        THROW_IF_FAILED(m_hostConfig->get_Spacing(&spacingConfig));
+
+        UINT32 cardPadding;
+        THROW_IF_FAILED(spacingConfig->get_Padding(&cardPadding));
+
+        double negativeCardMargin = cardPadding * -1.0;
```

Comment:
>double negativeCardMargin = cardPadding * -1.0; [](start = 8, length = 47)

[nit] Could you declare this right before usage?

Reply:
Fixed